File: progress_bar_v2.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>		//usleep()

#define BAR "||||||||||||||||||||"
#define BAR_WIDTH 20

void progress_bar2_test(unsigned int percentage)
{
	unsigned int lbar=percentage*BAR_WIDTH/100;
	unsigned int rbar=BAR_WIDTH-lbar;
	printf("\r[%.*s%*s] %d%% Completed.", lbar, BAR, rbar, "", percentage);
	fflush(stdout);
}

int main()
{
    unsigned int percentage=0;
	while(percentage<100){
		++percentage;
		progress_bar2_test(percentage);
		usleep(100000);
	}
	printf("\n");

	return 0;
}

File: progress_bar_v2.py
import sys
import time

BAR = "||||||||||||||||||||"
BAR_WIDTH = 20

def progress_bar2(percentage):
    lbar = percentage*BAR_WIDTH/100
    rbar = BAR_WIDTH-lbar
    print "\r[%.*s%*s] %d%% completed." % (lbar, BAR, rbar, "", percentage),
    sys.stdout.flush()

for i in range(101):
    progress_bar2(i)
    time.sleep(0.1)
